src/iexamine.cpp

```diff
@@ -4158,6 +4158,77 @@ static int get_charcoal_charges( units::volume food )
     return  std::max( charcoal, sm_rack::MIN_CHARCOAL );
 }
 
+static bool is_non_rotten_crafting_component( const item &it )
+{
+    return is_crafting_component( it ) && !it.rotten();
+}
+
+void mill_activate( player &p, const tripoint &examp )
+{
+    const furn_id cur_mill_type = g->m.furn( examp );
+    furn_id next_mill_type = f_null;
+    if( cur_mill_type == f_wind_mill ) {
+        next_mill_type = f_wind_mill_active;
+    } else if( cur_mill_type == f_water_mill ) {
+        next_mill_type = f_water_mill_active;
+    } else {
+        debugmsg( "Examined furniture has action mill_activate, but is of type %s",
+                  g->m.furn( examp ).id() );
```

Comment:
Sorry, you totally do need `c_str()` here
```suggestion
                  g->m.furn( examp ).id().c_str() );
```